platform/ios/src/MGLAnnotationView.mm

```diff
@@ -37,6 +38,25 @@ - (void)setCenterOffset:(CGVector)centerOffset
     self.center = self.center;
 }
 
+- (void)setSelected:(BOOL)selected
+{
+    [self setSelected:selected animated:NO];
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated
+{
+    [self willChangeValueForKey:@"selected"];
+    _selected = selected;
+    [self didChangeValueForKey:@"selected"];
+}
+
+- (void)setEnabled:(BOOL)enabled
+{
+    [self willChangeValueForKey:@"enabled"];
```

Comment:
`-willChangeValueForKey:`/`-didChangeValueForKey:` is only necessary when you set the ivar in a method other than a standard setter. So in the case of `-setSelected:animated:`, you need those calls, because it isn’t called `-setSelected:`.

Moreover, it’s unnecessary to implement `-setEnabled:` just to set the ivar; Objective-C can synthesize this method for you automatically.